### PR TITLE
[Intel MKL] Fixes for unit test failures

### PR DESCRIPTION
**Pull request by nhasabni**
1) Changes in partitioned_function_ops.cc are for passing
   Global OpRegistry as default_registry in PartitionedFunction op

   This fix addresses failure in MKL layout pass when PartitionedFunction
   op calls graph optimization passes. The problem was that the function
   library definition that is used to create function graph and corresponding
   subgraphs after partitioning did not use global OpRegistry as the
   default OpRegistry used for look of operator names. Because of that,
   standard operators such as "Const" were not available to graph passes.

2) Changes in mkl_cpu_allocator.h are to address failure in
   mkl_cpu_allocator_test which was expecting that max_bytes_limits is returned
   via GetStats() in MKLCPUAllocator.

### CODE REVIEW

**Comment by agramesh1** [comment]
@tatianashp @penpornk This PR fixes the failures in the MKL CI, thanks.

File: tensorflow/core/kernels/partitioned_function_ops.cc [hunk 1]
```diff
@@ -97,7 +97,12 @@ class PartitionedCallOp : public AsyncOpKernel {
         OP_REQUIRES_ASYNC(ctx, fbody != nullptr,
                           errors::Internal("Could not find handle ", handle),
                           done);
-        auto graph = tensorflow::MakeUnique<Graph>(fbody->graph->flib_def());
+        // We need to pass global op_registry as default_registry when creating
+        // graph. So that graph optimization passes can lookup all possible ops
+        // by name.
+        FunctionLibraryDefinition func_lib_def(OpRegistry::Global(),
+                                            fbody->graph->flib_def().ToProto());
```
Comment (andydavis1) [inline tensorflow/core/kernels/partitioned_function_ops.cc]: Indent

**Review by andydavis1 — DISMISSED** [review]

**Review by tatianashp — DISMISSED** [review]

**Comment by tatianashp** [comment]
@nhasabni Thank you for fixing the test failure! 

Please fix the formatting issues so that we could get this PR in.

**Review by tatianashp — DISMISSED** [review]

**Comment by nhasabni** [comment]
@tatianashp I looked into the failing test -- //tensorflow/python/eager:function_test seems to fail in spite of my fix. Can you confirm?

**Comment by nhasabni** [comment]
@tatianashp Sorry, my bad. The test is failing because of my fix. Will fix it.

**Comment by tatianashp** [comment]
Sounds good. Looking forward to the fix. Let us know if you need feedback from TF Eager folks.

**Comment by nhasabni** [comment]
@tatianashp I think I will need some input to fix the issue. I am not sure if the TensorFlow core API exists for what I am looking for. I think @mrry and @akshayka have worked on relevant piece of code. So if they can comment, that would be great.

**The main issue that I was trying to address was**: some unit tests fail with MKL graph passes because these graph passes need the definition of standard operators (like Const) in order to rewrite graphs. The definitions of these standard operators are only available in `OpRegistry::Global`, and the code in `partitioned_function_ops.cc` was not passing `OpRegistry::Global()` to the graph passes. That is why my change was to pass `OpRegistry::Global()` while creating graph in `partitioned_function_ops.cc`.

But with my change, `python:eager_test` is failing because it cannot find a symbol `add_9n57JfTlx2g` (The precise error is ```graph.cc:379] Non-OK-status: ops_.LookUpOpDef(node->type_str    ing(), &op_def) status: Not found: Op type not registered add_9n57JfTlx2g```. 

I tried to debug `eager_test` failure. I see that, by default, `partitioned_function_ops.cc` passes `flib_def` and the `default_registry()` for that `flib_def` from the graph for the input function while creating a graph that is passed to the graph optimization passes. I feel that my change passes `OpRegistry::Global()` instead of `default_registry()` for input `flib_def` while creating the graph. And I think this is creating the problem in `.

I think the solution should allow us to create a graph using both `default_registry` from input `flib_def` and `OpRegistry::Global()`. But I do not think such API exists.

Thanks.

**Comment by alextp** [comment]
What I do not understand here is why something is in OpRegistry::Global() and not in default_registry, which should always be a superset of OpRegistry::Global.

**Comment by alextp** [comment]
If you insist on creating a function library definition with the global registry then you should make sure to register all functions which you need in that new function library definition. Not sure if there's an API to get the list of registered functions, but it should not be hard to get one.

**Comment by nhasabni** [comment]
@alextp Thanks for input. Yes, I also did not understand. I thought that default_registry should always include global registry. But that does not seem to be the case. Especially, in `nested_function_test` of `eager:function_test` top-level functions `flib_def`seems to be passed as `default_registry` to child function.

**Comment by tatianashp** [comment]
The problem appeared after https://github.com/tensorflow/tensorflow/commit/eb71a1a3afbbe21407b2149d7adc4efa9e557b24 commit that added graph optimization passes to PartitionedCallOp.

@akshayka - Could you help to figure out what to do here?

**Review by tatianashp — DISMISSED** [review]
Thanks for tackling this.

**Comment by nhasabni** [comment]
@tatianashp Not sure if XLA failure is known? If not, I can debug.

**Comment by penpornk** [comment]
@nhasabni I don't think it has been reported yet. Please feel free to go for it. :)

**Comment by penpornk** [comment]
Sorry, I was wrong. This is known but hasn't been fixed yet. I'll see what we can do.

**Comment by tatianashp** [comment]
@nhasabni  Most likely Windows Bazel and XLA failures are not due to your changes since the tests don't use --config=mkl.
@qlzh727  Do you know if the failures above are specific to this PR?

**Comment by nhasabni** [comment]
I ran the XLA test without my PR and without MKL. It fails in my setup also.

**Comment by qlzh727** [comment]
Let me kick off a new test, from the dashboard, the XLA test is green at the moment, we will see if the new round of test passes or not.

**Comment by agramesh1** [comment]
@qlzh727 can you merge this PR? the XLA test is failing in tests:exhaustive_f32_elementwise_op_test_cpu, seems to be unrelated to this PR. Other PRs are also failing this XLA test. Thanks.

**Comment by agramesh1** [comment]
@qlzh727 The PR is stuck in "ready to pull". Can you take a look? Thanks and sorry to bother you.

**Comment by qlzh727** [comment]
It is failing for the our internal test, and @tatianashp will probably take more actions for it.

**Comment by agramesh1** [comment]
@qlzh727 Thanks. @tatianashp let me know how we can help.

**Comment by penpornk** [comment]
In `partitioned_function_ops.cc`, could you please wrap `TF_CHECK_OK` around both your `AddFunctionLibrary` calls (and make sure the formatting is ok)?
Line 105,
```c++
TF_CHECK_OK(
            graph.get()->AddFunctionLibrary(global_flib.ToProto()));
```
Line 261,
```c++
TF_CHECK_OK(
          subgraph.get()->AddFunctionLibrary(global_flib.ToProto()));
```
Thank you!

**Comment by nhasabni** [comment]
Thanks @penpornk. Pls take a look.

**Review by penpornk — APPROVED** [review]
Thank you for the changes!

**Pull request merged**